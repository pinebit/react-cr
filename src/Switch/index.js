import React from 'react';

const Switch = ({ children, index }) => {
  const childrenCount = React.Children.count(children);
  if (index >= childrenCount) {
    return null;
  }

  return children[index];
};

Switch.propTypes = {
  index: React.PropTypes.number.isRequired
};

export default Switch;
