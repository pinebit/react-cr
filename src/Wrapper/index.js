import React from 'react';

const Wrapper = ({ children, wrapper, reverse, ...wrapperProps }) => {
  if (!children) {
    return null;
  }

  const useWrapper = React.Children.count(children) > 1 || wrapper;
  if (useWrapper) {
    const type = wrapper || 'div';
    const _children = reverse ? [...children].reverse() : children;
    return React.createElement(type, wrapperProps, _children);
  }

  return children;
};

Wrapper.propTypes = {
  wrapper: React.PropTypes.string,
  reverse: React.PropTypes.bool
};

export default Wrapper;
