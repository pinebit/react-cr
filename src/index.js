module.exports = {
  demo: require('./Visibility').default
};

