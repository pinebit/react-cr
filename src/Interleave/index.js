import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from '../Wrapper';

const Interleave = ({ separator, first, last, children, ...wrapperProps }) => {
  if (!children || !separator) {
    return null;
  }

  const content = [];
  children.forEach((child, index) => {
    const key = child.key;
    if (key) {
      content.push(child);
      const lastChild = index === children.length - 1;
      if (!lastChild) {
        content.push(React.cloneElement(separator, { key: `${key}_separator` }));
      }
    }
  });

  return (
    <Wrapper {...wrapperProps}>
      {first && React.cloneElement(separator, { key: 'first_separator' })}
      {content}
      {last && React.cloneElement(separator, { key: 'last_separator' })}
    </Wrapper>
  );
};

Interleave.propTypes = {
  separator: PropTypes.node.isRequired,
  first: PropTypes.bool,
  last: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default Interleave;
