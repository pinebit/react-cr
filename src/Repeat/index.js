import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from '../Wrapper';

const Repeat = ({ count, children, ...wrapperProps }) => {
  if (!children || count <= 0 || Array.isArray(children)) {
    return null;
  }

  if (count === 1) {
    return (
      <Wrapper {...wrapperProps}>
        {children}
      </Wrapper>
    );
  }

  const fakeArray = Array.from(new Array(count)).map((x, index) => index + 1);

  return (
    <Wrapper {...wrapperProps}>
      {fakeArray.map(key => React.cloneElement(children, { key }))}
    </Wrapper>
  );
};

Repeat.propTypes = {
  count: PropTypes.number.isRequired,
  children: PropTypes.node
};

export default Repeat;
