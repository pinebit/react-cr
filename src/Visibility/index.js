import React from 'react';
import Wrapper from '../Wrapper';

const Visibility = ({ visible, ...wrapperProps }) => {
  if (!visible) {
    return null;
  }

  return <Wrapper {...wrapperProps} />;
};

Visibility.defaultProps = {
  visible: true
};

Visibility.propTypes = {
  visible: React.PropTypes.bool
};

export default Visibility;
