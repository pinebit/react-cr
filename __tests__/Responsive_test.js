import React from 'react';
import renderer from 'react-test-renderer';
import { Desktop, Mobile } from '../src/Responsive';

it('Desktop renders correctly with implicit wrapping', () => {
  const tree = renderer.create(
    <Desktop>
      <h1>1</h1>
      <h1>2</h1>
    </Desktop>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Desktop renders correctly with header wrapper', () => {
  const tree = renderer.create(
    <Desktop wrapper="header">
      <h1>1</h1>
    </Desktop>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Desktop renders correctly without wrapping', () => {
  const tree = renderer.create(
    <Desktop>
      <h1>1</h1>
    </Desktop>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Mobile does not render because we are on desktop', () => {
  const tree = renderer.create(
    <Mobile>
      <h1>1</h1>
      <h1>2</h1>
      <h1>3</h1>
    </Mobile>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
