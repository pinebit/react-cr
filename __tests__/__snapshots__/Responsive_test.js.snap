// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop renders correctly with header wrapper 1`] = `
<header>
  <h1>
    1
  </h1>
</header>
`;

exports[`Desktop renders correctly with implicit wrapping 1`] = `
<div>
  <h1>
    1
  </h1>
  <h1>
    2
  </h1>
</div>
`;

exports[`Desktop renders correctly without wrapping 1`] = `
<h1>
  1
</h1>
`;

exports[`Mobile does not render because we are on desktop 1`] = `null`;
